### PR TITLE
TEST/UCP: Meet aligned_alloc requirements for requested size

### PR DESCRIPTION
**Pull request by ivankochin**
## What
Make message size divisible to page size.

## Why ?
`aligned_alloc` requires that allocated size should be divisible to alignment

[cppreference](https://en.cppreference.com/w/c/memory/aligned_alloc):

> Allocate size bytes of uninitialized storage whose alignment is specified by alignment. The size parameter must be an integral multiple of alignment.

Found by ASAN.

### CODE REVIEW

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 1]
```diff
@@ -248,7 +248,13 @@ size_t test_ucp_tag_xfer::get_msg_size()
 {
     size_t raw_size   = 1148544 / ucs::test_time_multiplier();
     size_t chunk_size = num_lanes() * UCS_SYS_PCI_MAX_PAYLOAD;
-    return ucs_div_round_up(raw_size, chunk_size) * chunk_size;
+    size_t page_size  = ucs_get_page_size();
+
+    EXPECT_TRUE(ucs_is_pow2(chunk_size) && ucs_is_pow2(page_size)) <<
+                "message size should be aligned to both page and chunk sizes";
+    size_t alignment  = ucs_max(chunk_size, page_size);
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:255]: I think chunk_size may be not a power of 2 - if num_lanes is not a power of 2
maybe `alignment = ucs_max(page_size, UCS_SYS_PCI_MAX_PAYLOAD) * num_lanes()`

Comment (ivankochin) [reply test/gtest/ucp/test_ucp_tag_xfer.cc]: Done.

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 2]
```diff
@@ -246,9 +246,11 @@ void test_ucp_tag_xfer::test_xfer_prepare_bufs(uint8_t *sendbuf, uint8_t *recvbu
 
 size_t test_ucp_tag_xfer::get_msg_size()
 {
-    size_t raw_size   = 1148544 / ucs::test_time_multiplier();
-    size_t chunk_size = num_lanes() * UCS_SYS_PCI_MAX_PAYLOAD;
-    return ucs_div_round_up(raw_size, chunk_size) * chunk_size;
+    size_t raw_size  = 1148544 / ucs::test_time_multiplier();
+    size_t alignment = ucs_max(UCS_SYS_PCI_MAX_PAYLOAD, ucs_get_page_size()) *
```
Comment (brminich) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:250]: `aligned_alloc` below uses just `ucs_get_page_size` as an alignment

File: test/gtest/ucp/test_ucp_tag_xfer.cc [hunk 3]
```diff
@@ -246,9 +246,9 @@ void test_ucp_tag_xfer::test_xfer_prepare_bufs(uint8_t *sendbuf, uint8_t *recvbu
 
 size_t test_ucp_tag_xfer::get_msg_size()
 {
-    size_t raw_size   = 1148544 / ucs::test_time_multiplier();
-    size_t chunk_size = num_lanes() * UCS_SYS_PCI_MAX_PAYLOAD;
-    return ucs_div_round_up(raw_size, chunk_size) * chunk_size;
+    size_t raw_size  = 1148544 / ucs::test_time_multiplier();
```
Comment (ivankochin) [inline test/gtest/ucp/test_ucp_tag_xfer.cc:249]:
```suggestion
    size_t raw_size = 1148544 / ucs::test_time_multiplier();
```

**Review by yosefe — APPROVED** [review]

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]
pls squash

**Pull request merged**